okio/src/commonMain/kotlin/okio/internal/-Path.kt

```diff
@@ -159,6 +155,84 @@ internal inline fun Path.commonResolve(child: Path): Path {
   return buffer.toPath()
 }
 
+@ExperimentalFileSystem
+@Suppress("NOTHING_TO_INLINE")
+internal inline fun Path.commonRelativeTo(other: Path): Path {
+  require(isAbsolute && other.isAbsolute || isRelative && other.isRelative) {
+    "Paths of different roots cannot be relative to each other: $this and $other"
+  }
+
+  val thisSlash = slash
+  val otherSlash = other.slash
+  require(thisSlash == otherSlash) {
```

Comment:
I think this is too strict for null slashes. I think these should pass?

```
val a = "foo".toPath()
val b = "foo/bar".toPath()
assertEquals(a.relativeTo(b), "..".toPath())
assertEquals(b.relativeTo(a), "bar".toPath())
```

I also think we should forgive different slashes if they aren’t leading:

```
val a = "foo/red".toPath()
val b = "foo\\green".toPath()
assertEquals(a.relativeTo(b), "...\\red".toPath())
assertEquals(b.relativeTo(a), "../green".toPath())
```